src/morphodict/analysis/tag_map.py

```diff
@@ -0,0 +1,166 @@
+class UnknownTagError(KeyError):
+    """Raised when TagMap encounters an unknown tag"""
+
+
+class MultiTagCopyError(Exception):
+    """
+    Raised when multiple tags are present but not expected
+    """
+    pass
+
+
+class MultiplePrecedenceError(Exception):
+    """
+    Raised when a single tag has multiple precedences declared
+    """
+    pass
+
+
+class ConflictingPrecedenceError(Exception):
+    """
+    Raised when multiple tags have the same precedence
+    """
+    pass
+
```

Comment:
These exceptions should all inherit from some common InvalidTagMapException.

Because they are so specific, calling code could also be simplified by pre-filling the message in the constructor, but that is optional.